internal/provider/models/modelutils/conditional_types.go

```diff
@@ -125,3 +128,126 @@ func GetAttributesByLevel(level int) map[string]schema.Attribute {
 func GetChildExpressionGroupTypeByLevel(level int) attr.Type {
 	return childExpressionGroups[level].Type()
 }
+
+func UnwindConditionalToModel(component map[string]any) ObjectValue {
```

Comment:
Moved out of reduce processor and made a couple of modifications to ensure valid HCL is generated

Reply:
Thanks!  LGTM